File: myauthclient.py
import sys
sys.path.append('gen-py')

from myauth import Authentication
from myauth.constants import *

from thrift import Thrift
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol

transport = TSocket.TSocket('localhost', 9090)
transport = TTransport.TBufferedTransport(transport)
protocol = TBinaryProtocol.TBinaryProtocol(transport)

auth = Authentication.Client(protocol)

transport.open()

print auth.say_hello()

#user = User()
#user.username = 'hello'
#user.password = 'world'
#
#print "Login: %s" % auth.login(user)
#
#user2 = User()
#user2.username = 'failed'
#user2.password = 'world'
#
#print "Login: %s" % auth.login(user2)

File: myauthclient.rb
#!/usr/bin/env ruby1.9

require 'thrift'
$:.push('gen-rb')

require 'authentication'
require 'myauth_constants'

transport = Thrift::BufferedTransport.new(Thrift::Socket.new('localhost', 9090))
protocol = Thrift::BinaryProtocol.new(transport)
auth = MyAuth::Authentication::Client.new(protocol)
transport.open()

puts auth.say_hello()

user = MyAuth::User.new
user.username = 'hello'
user.password = 'world'

puts "Login: #{auth.login(user)}"

user2 = MyAuth::User.new
user2.username = 'failed'
user2.password = 'world'

puts "Login: #{auth.login(user2)}"
